pkg/extractor/index.go

```diff
@@ -0,0 +1,36 @@
+package extractor
+
+import (
+	"context"
+	"sort"
+
+	"github.com/mt-sre/addon-metadata-operator/pkg/types"
+	"github.com/operator-framework/operator-registry/alpha/action"
+)
+
+type indexExtractor struct {
+	types.IndexExtractorCache
+}
+
+func NewIndexExtractor(cache types.IndexExtractorCache) types.IndexExtractor {
+	return indexExtractor{cache}
+}
+
+// ListBundles - return a list of all bundle images, sorted
+func (i indexExtractor) ListBundles(indexImage string, packageName string) ([]string, error) {
+	if bundles := i.GetBundles(indexImage, packageName); bundles != nil {
+		return bundles, nil
+	}
+	res := make([]string, 0)
+	lb := action.ListBundles{IndexReference: indexImage, PackageName: packageName}
```

Comment:
Just wondering how we'd know the package names of all dependent operators in an indeximage.

Reply:
So if we leave `packageName` to "", it lists bundles for all packages. So we can always add this functionality. I did not for this PR as I was wondering in what cases do we need to know the bundles from dependent operators? (e.g.: prometheus) Do we have a need/use case for it?

My understanding is that we only need to list bundles associated with the `operatorName` field from AddonMetadata (which maps to the `packageName`), so we can cross validate metadata and bundles.

Reply:
>Do we have a need/use case for it?
`mtcli list-bundles` should list all the operators in the image right?

Reply:
That's right, ok let me make it obvious `ListBundles` vs `ListAllBundles`